### DISCUSSION TITLE
Delete Eventrule where the script does not exist anymore

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#16912] [Help Wanted!] [answered]
I have the following problem:

I cannot delete event rules where the object (script) no longer exists. 
```
<class 'AttributeError'>

'ScriptModule' object has no attribute 'python_class'
```

How can I delete them? Same error when trying to edit them. I think this happened when upgrading to v4 because they now are assigned in some other way.

**Comment by PaulR282** [comment]
Looks like the update broke it because the relation to the script no longer works. If I add a new one with the same script, the object relation works (I can click on the object link, the old broken ones go to `/extras/scripts` instead of `extras/scripts/[id]`.

**Comment by jeremystretch** [comment]
If you're able to reproduce this reliably, please open a bug report and we'll work to resolve it.

**Reply by PaulR282** [reply]
I could maybe reproduce this when migrating in instance from v3 to v4. Same problem: #16326. I was already able to resolve it on the database.

**Reply by jhofmueller** [reply]
It looks like this bug has been solved but the same thing happens to webhooks after upgrading from 3.7.8 to 4.0.9.

**Comment by PaulR282** [comment, marked as answer]
Solution: 
I truncated the tables `extras_eventrule_object_types` and `extras_eventrule` on the database. This deletes all event rules.

**Reply by chris240189** [reply]
> Solution: I truncated the tables `extras_eventrule_object_types` and `extras_eventrule` on the database. This deletes all event rules.

Thanks, that helped me a lot! I had the same issue with an event rule after upgrading to v4.0.8 from v3.7.8
